xcdat/bounds.py

```diff
@@ -138,98 +139,109 @@ def add_missing_bounds(self, width: float = 0.5) -> xr.Dataset:
         -------
         xr.Dataset
         """
-        axes = CF_NAME_MAP.keys()
+        axes = CF_ATTR_MAP.keys()
 
         for axis in axes:
-            # Check if the axis coordinates can be mapped to.
             try:
-                get_axis_coord(self._dataset, axis)
+                coords = get_axis_coord(self._dataset, axis)
             except KeyError:
                 continue
 
-            # Determine if the axis is also a dimension by determining if there
-            # is overlap between the CF axis names and the dimension names. If
-            # not, skip over axis for validation.
-            if len(set(CF_NAME_MAP[axis]) & set(self._dataset.dims.keys())) == 0:
-                continue
-
-            # Check if bounds also exist using the "bounds" attribute.
-            # Otherwise, try to add bounds if it meets the function's criteria.
-            try:
-                self.get_bounds(axis)
-                continue
-            except KeyError:
-                pass
+            for coord in coords.coords.values():
+                try:
+                    self.get_bounds(axis, str(coord.name))
+                    continue
+                except KeyError:
+                    pass
```

Comment:
I think we might be able to only add bounds to `dims` to resolve the `height`/singleton issue. This block would become:

```
            for coord in coords.coords.values():
                if coord.name not in self._dataset.dims:
                    continue
                try:
                    self.get_bounds(axis, str(coord.name))
                    continue
                except KeyError:
                    pass
```

Reply:
Actually, we should keep this logic to add bounds for all non-dimensional coordinates that are not singleton.

We just need to update the conditional which assumes singletons are always 1-d arrays (they can be 0-d arrays too).
My comment here with the solution: https://github.com/xCDAT/xcdat/pull/343#discussion_r994969579

Reply:
I don't think we need bounds for non-dimensional axes, but if we can add them without creating problems, that should be okay.